spec/requests/requests_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe 'managing topics', :type => :request do
+
+  let!(:user) {FactoryBot.create :user}
+
+  it 'redirects unauthenticated users' do
+    post "/topics", :params => { :topic => {:name => "here is a new topic"} }
+    expect(response).to redirect_to('/users/sign_in')
+  end
+
+  it 'allows topic posting for logged in users' do
+    post user_session_path, :params => {'user[email]' => user.email, 'user[password]' => user.password }
```

Comment:
Don't do user authentication like this (unless it is the thing you are testing - which it isn't here).

Instead use warden test helpers provided by Devise. See [this](https://github.com/plataformatec/devise/wiki/How-To:-Test-with-Capybara) for some clues - it's talking about capybara, which isn't relevant, but the helpers should work for request specs.